linera-core/src/updater.rs

```diff
@@ -380,12 +380,24 @@ where
                     )
                     .await?;
                 }
-                Err(NodeError::MissingCrossChainUpdate { .. }) if !sent_cross_chain_updates => {
+                Err(NodeError::MissingCrossChainUpdate {
+                    chain_id,
+                    origin,
+                    height,
+                }) if chain_id == proposal.content.block.chain_id
+                    && sent_cross_chain_updates
+                        .get(&origin)
+                        .is_none_or(|h| *h < height) =>
```

Comment:
If `height < h` (i.e. there's a block on `origin` at height `height` is sending a message to `chain_id` that this validator hasn't received yet) – shouldn't we send those blocks anyway?

Reply:
`send_chain_information` sends all blocks from the sender chain that we have, up to the given height. So if `h > height` here, that means we have already sent them everything we have, up to a higher height. No need to try again for now.

Reply:
Yes, I realized that after I wrote it but you were faster.